tests/integration/test_backup_restore_on_cluster/test_disallow_concurrency.py

```diff
@@ -111,69 +111,84 @@ def create_and_fill_table():
         nodes[i].query(f"INSERT INTO tbl SELECT number FROM numbers(40000000)")
 
 
-def wait_for_fail_backup(node, backup_id, backup_name):
+def get_status_and_error(node, backup_or_restore_id):
+    return (
+        node.query(
+            f"SELECT status, error FROM system.backups WHERE id == '{backup_or_restore_id}'"
+        )
+        .rstrip("\n")
+        .split("\t")
+    )
+
+
+def wait_for_backup(node, backup_id):
+    assert_eq_with_retry(
+        node,
+        f"SELECT status FROM system.backups WHERE id = '{backup_id}'",
+        "BACKUP_CREATED",
+        sleep_time=2,
+        retry_count=50,
+    )
+
+
+def wait_for_restore(node, restore_id):
+    assert_eq_with_retry(
+        node,
+        f"SELECT status FROM system.backups WHERE id == '{restore_id}'",
+        "RESTORED",
+        sleep_time=2,
+        retry_count=50,
+    )
+
+
+def check_backup_error(error):
     expected_errors = [
         "Concurrent backups not supported",
-        f"Backup {backup_name} already exists",
+        "BACKUP_ALREADY_EXISTS",
```

Comment:
The main part of the fix is here. The test failed sometimes because it expected error
```
Exception(ErrorCodes::BACKUP_ALREADY_EXISTS, "Backup {} already exists", ...);
```
whileas the code sometimes throws also error
```
Exception(ErrorCodes::BACKUP_ALREADY_EXISTS, "Backup {} is being written already", ...)
```
So it seems it's more reliable to check for error code BACKUP_ALREADY_EXISTS instead.